hal/mbed_ticker_api.c

```diff
@@ -375,7 +375,7 @@ void ticker_insert_event_us(const ticker_data_t *const ticker, ticker_event_t *o
     obj->next = p;
 
     /* if prev is NULL we're at the head */
-    if (prev == NULL) {
+    if (prev == NULL || timestamp <= ticker->queue->present_time) {
```

Comment:
I just noticed that this implementation is wrong. Should be something like:

```
    if (prev == NULL) {
        ticker->queue->head = obj;
    } else {
        prev->next = obj;
    }

    if (prev == NULL || timestamp <= ticker->queue->present_time) {
        schedule_interrupt(ticker);
    }
```


I will test this tomorrow and provide the fix.

Reply:
Fixed